compiler/src/iree/compiler/InputConversion/StableHLO/Preprocessing/StableHLOToStableHLO.cpp

```diff
@@ -1540,6 +1540,71 @@ struct CustomCallIsTopK final
   }
 };
 
+struct IotaSortSliceIsTopK final : OpRewritePattern<mlir::stablehlo::SortOp> {
+  using OpRewritePattern::OpRewritePattern;
+
+  LogicalResult matchAndRewrite(mlir::stablehlo::SortOp op,
+                                PatternRewriter &rewriter) const override {
+    auto opOperands = op.getOperands();
+    auto opResults = op.getResults();
+    Value topKInput;
+    if (opOperands.size() != 2 || opResults.size() != 2) {
+      return rewriter.notifyMatchFailure(
+          op, "Slice that maps to TopK must have exactly two inputs/outputs");
+    }
+
```

Comment:
Note, something I missed. We will need to check the `mhlo::SortOp`'s region, such that it includes the right type of comparison.  You can see a similar check in https://github.com/openxla/iree/pull/13937

Reply:
Done. Just to verify, LT shouldn't work then? (I'm guessing it would be "BottmK" in that case...)